app/src/main/java/com/infomaniak/swisstransfer/ui/components/transfer/TransferItem.kt

```diff
@@ -0,0 +1,170 @@
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun TransferItem(
+    transfer: TransferUi,
+    onClick: () -> Unit,
+) {
+    Card(
+        onClick = onClick,
+        colors = CardDefaults.cardColors(containerColor = SwissTransferTheme.materialColors.surfaceContainerHighest),
+        shape = CustomShapes.SMALL,
+    ) {
+        Row(
+            modifier = Modifier.padding(Margin.Medium),
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+
+            Column(
+                modifier = Modifier.weight(1.0f),
+            ) {
+
+                Text(
+                    text = Date(transfer.createdDateTimestamp).format(FORMAT_DATE_TITLE),
+                    style = SwissTransferTheme.typography.bodyMedium,
+                    color = SwissTransferTheme.colors.primaryTextColor,
+                    maxLines = 1,
+                    overflow = TextOverflow.MiddleEllipsis,
+                )
+
+                Spacer(modifier = Modifier.height(Margin.Mini))
+                TextDotText(
+                    firstText = { Formatter.formatShortFileSize(LocalContext.current, transfer.sizeUploaded) },
+                    secondText = { stringResource(R.string.expiresIn, transfer.expiresInDays) },
+                )
+
+                Spacer(modifier = Modifier.height(Margin.Mini))
+                ContextualFlowRow(
+                    itemCount = transfer.files.count(),
+                    maxLines = 1,
+                    overflow = ContextualFlowRowOverflow.expandIndicator { TransferFilePreview(remainingFilesCount = totalItemCount - shownItemCount) },
+                ) { index ->
+                    TransferFilePreview(
+                        file = transfer.files[index],
+                        isFirstItem = index == 0,
+                    )
+                }
+            }
+
+            Spacer(modifier = Modifier.width(Margin.Medium))
+            Icon(
+                imageVector = AppIcons.ChevronRightThick,
+                contentDescription = null,
+                modifier = Modifier.size(Margin.Medium),
+                tint = SwissTransferTheme.colors.iconColor,
+            )
+        }
+    }
+}
+
+@PreviewAllWindows
+@Composable
+private fun Preview() {
+    SwissTransferTheme {
+        Surface {
+            TransferItem(
+                transfer = TransferUi(
+                    uuid = UUID.randomUUID().toString(),
+                    createdDateTimestamp = Date().time - 30L * 86_400_000L,
+                    expirationDateTimestamp = Calendar.getInstance().apply {
+                        time = Date()
+                        set(Calendar.DATE, get(Calendar.DATE) + 1)
+                    }.time.time,
+                    sizeUploaded = 57_689_032L,
+                    downloadLimit = 10,
+                    downloadLeft = 8,
+                    message = "Coucou c'est moi le message de description du transfert.",
+                    files = listOf(
```

Comment:
Maybe put this files "computation" in a variable out of the composable, to make it a bit more readable

Reply:
That's hardcoded temporary code, we don't really care about it.

Reply:
That's a preview, that's not temporary 
And you already did that in another file so it logical to do it there to

Reply:
Oh my bad, that's the Preview one, right.